### ISSUE TITLE
[Star info] Star discovery infobox never disappears

### ISSUE THREAD

**Issue by emiljensen2** [pwatts6060/runelite-plugins#35]
The bonus stardust infobox never disappears if u don't mine the star, it just stays on screen.
Leaving the area/tp/etc doesn't clear it, world hop doesn't clear it, switching the plugin off and on doesn't clear it, logging out and in again doesn't clear it. A new star cycle doesn't clear it.
You have to restart your client to make it disappear.

**Comment by pwatts6060** [comment]
What conditions do you think should it be cleared? When the player no longer has pickaxe on them? People changing locations/worlds to go to another star probably don't want it cleared.

**Comment by emiljensen2** [comment]
Yeah no pickaxe is an decent idea and varbit IN_RAID andor instance.
Tough I guess this wouldn't cover edge cases like zalcano/mlm, questing. Not sure if one could perhaps store a variable of what was last mined, and if it wasn't a star one could conclude they're doing other things.

**Comment by Koslas1** [comment]
I have had the bonus stardust on my screen for almost a week now. Closing the client and opening it again doesn't seem to remove it. I have just gotten annoyed at it being constantly there, and found a random star to mine, which is making it remaining stardust go down.

My suggestion would be if possible to store the time of when you discover the star somewhere, and if it has been more than 2 hours since that time, to not show the bonus infobox. When you are the first to discover a star, then overwrite the stored time, and since it would be less than 2 hours since discovering it again, the infobox appears again. I'm not sure if that's possible though. 
The reasoning for the 2 hours is as the wiki states this "A shooting star will fall at a random spot selected from a predetermined list roughly every 90 minutes (with a variation of up to 15 minutes)" And if it has the variation of 15 minutes later, then that's 105 minutes (1 hour and 45 minutes) before the next star falls, and then you won't get the bonus stardust, unless you discover the next star.

**Comment by pwatts6060** [comment]
> I have had the bonus stardust on my screen for almost a week now. Closing the client and opening it again doesn't seem to remove it. I have just gotten annoyed at it being constantly there, and found a random star to mine, which is making it remaining stardust go down.
> 
> My suggestion would be if possible to store the time of when you discover the star somewhere, and if it has been more than 2 hours since that time, to not show the bonus infobox. When you are the first to discover a star, then overwrite the stored time, and since it would be less than 2 hours since discovering it again, the infobox appears again. I'm not sure if that's possible though. The reasoning for the 2 hours is as the wiki states this "A shooting star will fall at a random spot selected from a predetermined list roughly every 90 minutes (with a variation of up to 15 minutes)" And if it has the variation of 15 minutes later, then that's 105 minutes (1 hour and 45 minutes) before the next star falls, and then you won't get the bonus stardust, unless you discover the next star.

A user defined cooldown of not discovering a star plus no pickaxe in inventory might be the way to go, ty for the suggestion

**Comment by pwatts6060** [comment]
https://github.com/runelite/plugin-hub/pull/7955 Should be fixed with this update

**Comment by Koslas1** [comment]
How does it work if I discover a star without a pickaxe in my inventory though? As I sometimes do that and then go grab my pickaxe and mine it, from when I'm just teleporting around doing other things? As I would want the Infobox to stay while I'm going to get my pickaxe?

Also you mention no pickaxes in the inventory, does that include checking pickaxes that you're welding?

**Comment by pwatts6060** [comment]
> How does it work if I discover a star without a pickaxe in my inventory though? As I sometimes do that and then go grab my pickaxe and mine it, from when I'm just teleporting around doing other things? As I would want the Infobox to stay while I'm going to get my pickaxe?
> 
> Also you mention no pickaxes in the inventory, does that include checking pickaxes that you're welding?

I didn't do anything with pickaxes. Instead it's a 10 minute timer (can edit to w/e amount of minutes) since you encountered a star. So you can find a star, leave it, world hop, the info box should stay there for 10 minutes. By find a star I mean be in the vicinity of it. So if you a find a star, the box will show up, and will still be shown when you grab the pickaxe from the bank and return. The timer starts ticking down once a star is no longer visible.

**Comment by pwatts6060** [comment]
This fix should be live now
